File: lista.1/batalha.naval/Jogador.java
import java.util.Scanner;

class Jogador {
  private String nome;
  private Tabuleiro tabuleiro;

  public Jogador(String nome, Tabuleiro tabuleiro) {
    this.nome = nome;
    this.tabuleiro = tabuleiro;
  }

  public String getNome() {
    return nome;
  }

  public void jogar() {
    Scanner scanner = new Scanner(System.in);

    tabuleiro.exibir(false);

    int linha, coluna;
    System.out.println(this.nome+ ", enter hit row/column:");

    do{
      linha = scanner.nextInt();
      coluna = scanner.nextInt();
    } while(!tabuleiro.atirar(linha, coluna, nome));

    tabuleiro.exibir(false);
  }

  public boolean perdeu() {
    return tabuleiro.perdeu();
  }

  public void exibirTab() {
    tabuleiro.exibir(true);
  }
}

File: lista.2/batalha.naval/Jogador.py
from Tabuleiro import Tabuleiro

class Jogador:
    
    def __init__(self, nome, tabuleiro):
        self.nome = nome
        self.tabuleiro = tabuleiro
    
    def getNome(self):
        return self.nome
    
    def jogar(self):
        self.tabuleiro.exibir(False)

        print(f"{self.getNome()}, enter hit row/column: ", end="")

        while True:
            linha = int(input())
            coluna = int(input())
            if self.tabuleiro.atirar(linha, coluna, self.nome):
                break

        self.tabuleiro.exibir(False)
    
    def perdeu(self):
        return self.tabuleiro.perdeu()
    
    def exibirTab(self):
        self.tabuleiro.exibir(True)
